persistence/sql/persister.go

```diff
@@ -135,6 +136,56 @@ type node interface {
 	GetNID() uuid.UUID
 }
 
+func (p *Persister) CleanupDatabase(ctx context.Context, wait time.Duration, older time.Duration) error {
+	currentTime := time.Now().Add(-older)
```

Comment:
```suggestion
	removeOlderThan := time.Now().Sub(older)
```

Reply:
This will cause a type mismatch. Since we're working on small durations I would live it this way